### PR TITLE
Skyline: Improve transition list format detection

### PR DESCRIPTION
**Pull request by brendanx67**
- detect international formats regardless of the current culture

### CODE REVIEW

**Comment by brendanx67** [comment]
Added a new test to test all the various formats and ran that test in SkylineTester under all possible current cultures.

**Comment by brendanx67** [comment]
This should impact the SmallMoleculeTransitionListCSVReader also. Maybe the code in that file:
            // Skip over header line to deduce decimal format
            string line = csvText.Count > 1 ? csvText[1] : header;
            MassListImporter.IsColumnar(line, out formatProvider, out separator, out columnTypes);
            // Double check that separator - does it appear in header row, or was it just an unlucky hit in a text field?
            if (!header.Contains(separator))
            {
                // Try again, this time without the distraction of a plausible but clearly incorrect seperator
                MassListImporter.IsColumnar(line.Replace(separator,'_'), out formatProvider, out separator, out columnTypes);
            }

Should change to use MassListInputs

File: pwiz_tools/Skyline/Model/Import.cs [hunk 1]
```diff
@@ -362,8 +362,21 @@ private void InitFormat(IList<string> inputLines)
                 char sep;
                 IFormatProvider provider;
                 Type[] columnTypes;
-                if (!MassListImporter.IsColumnar(inputLines[0], out provider, out sep, out columnTypes))
+                string inputLine = 0 < inputLines.Count ? inputLines[0] : string.Empty;
+                if (!MassListImporter.IsColumnar(inputLine, out provider, out sep, out columnTypes))
+                {
                     throw new IOException(Resources.SkylineWindow_importMassListMenuItem_Click_Data_columns_not_found_in_first_line);
+                }
+                // If there are no numbers in the first column, try the second. Without numbers the format provider may not be correct
```
Comment (nickshulman) [inline pwiz_tools/Skyline/Model/Import.cs:370]: I think this comment supposed to say "first row" instead of "first column"

**Review by nickshulman — APPROVED** [review]
Looks good to me.

**Review by bspratt — APPROVED** [review]
Looks good. I like the improved modularity.

**Pull request merged**